src/class-cache-collector.php

```diff
@@ -7,9 +7,517 @@
 
 namespace Cache_Collector;
 
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use WP_Post;
+use WP_Term;
+
 /**
- * Example Plugin
+ * Cache Collector
  */
 class Cache_Collector {
+	/**
+	 * Cache type for the object cache.
+	 *
+	 * @var string
+	 */
+	public const CACHE_OBJECT_CACHE = 'cache';
+
+	/**
+	 * Cache type for the transient cache.
+	 *
+	 * @var string
+	 */
+	public const CACHE_TRANSIENT = 'transient';
+
+	/**
+	 * Delimiter for the cache key.
+	 *
+	 * @var string
+	 */
+	public const DELIMITER = '_:_';
+	/**
+	 * Post type for storage.
+	 *
+	 * @var string
+	 */
+	public const POST_TYPE = 'cache_collector';
+
+	/**
+	 * Meta key for storage.
+	 *
+	 * @var string
+	 */
+	public const META_KEY = 'cache_collector_keys';
+
+	/**
+	 * Threshold in seconds for purging the cache related to a post when a post
+	 * is updated.
+	 *
+	 * Defaults to 5 days.
+	 *
+	 * @var int
+	 */
+	public static int $post_update_threshold = 432000;
+
+	/**
+	 * Default threshold for a cache key to expire and be removed from the cache
+	 * collector.
+	 *
+	 * @var integer
+	 */
+	public static int $expiration_threshold = 432000;
```

Comment:
Could you explain a little more what the purpose of this threshold is? Is it my responsibility as a developer to make sure the keys registered with the plugin expire in coordination with the actual TTLs on the cached values?

Reply:
It's killed